comp/otelcol/otlp/integrationtest/integration_test.go

```diff
@@ -119,8 +119,8 @@ func runTestOTelAgent(ctx context.Context, params *subcommands.GlobalParams) err
 		}),
 		logsagentpipelineimpl.Module(),
 		// We create strategy.ZlibStrategy directly to avoid build tags
-		fx.Provide(strategy.NewZlibStrategy),
-		fx.Provide(func(s *strategy.ZlibStrategy) compression.Component {
+		fx.Provide(compressionimpl.NewComponent),
+		fx.Provide(func(s *compressionimpl.ZlibStrategy) compression.Component {
```

Comment:
Actually, both are required, but I think this needs to be the following, given that `compressionimpl.NewComponent` returns `Provides`. 

```suggestion
		fx.Provide(func(s *compression.Provides) compression.Component {
					return s.Comp
```